backend/src/services/certificate-est/certificate-est-service.ts

```diff
@@ -0,0 +1,271 @@
+              .replace("-----END CERTIFICATE-----", "")
+              .replace(/\n/g, "")
+              .replace(/ /g, "")
+              .trim();
+
+            const certificateBuffer = Buffer.from(processedBody, "base64");
+            return new x509.X509Certificate(certificateBuffer);
+          })
+      );
+
+    if (!parsedChains || !parsedChains.length) {
+      throw new BadRequestError({
+        message: "Error parsing CA chain"
+      });
+    }
+
+    const certValidityAgainstChains = await Promise.all(
+      parsedChains.map(async (chain) => {
+        if (!chain) {
+          return false;
+        }
+
+        return checkCertValidityAgainstChain(cert, chain);
+      })
+    );
+
+    if (certValidityAgainstChains.every((isCertValid) => !isCertValid)) {
+      throw new BadRequestError({
+        message: "Invalid client certificate"
+      });
+    }
+
+    // We ensure that the Subject and SubjectAltNames of the CSR and the existing certificate are exactly the same
+    const csrObj = new x509.Pkcs10CertificateRequest(csr);
+    if (csrObj.subject !== cert.subject) {
+      throw new BadRequestError({
+        message: "Subject mismatch"
+      });
+    }
+
+    let csrSanSet: Set<string> = new Set();
+    const csrSanExtension = csrObj.extensions.find((ext) => ext.type === "2.5.29.17");
+    if (csrSanExtension) {
+      const sanNames = new x509.GeneralNames(csrSanExtension.value);
+      csrSanSet = new Set([...sanNames.items.map((name) => `${name.type}-${name.value}`)]);
+    }
+
+    let certSanSet: Set<string> = new Set();
+    const certSanExtension = cert.extensions.find((ext) => ext.type === "2.5.29.17");
+    if (certSanExtension) {
+      const sanNames = new x509.GeneralNames(certSanExtension.value);
+      certSanSet = new Set([...sanNames.items.map((name) => `${name.type}-${name.value}`)]);
+    }
+
+    if (csrSanSet.size !== certSanSet.size || ![...csrSanSet].every((element) => certSanSet.has(element))) {
+      throw new BadRequestError({
+        message: "Subject alternative names mismatch"
+      });
+    }
+
+    const { rawCertificate } = await certificateAuthorityService.signCertFromCa({
+      isInternal: true,
+      certificateTemplateId,
+      csr
+    });
+
+    return convertRawCertsToPkcs7([rawCertificate]);
+  };
+
+  const simpleEnroll = async ({
+    csr,
+    certificateTemplateId,
+    sslClientCert
+  }: {
+    csr: string;
+    certificateTemplateId: string;
+    sslClientCert: string;
+  }) => {
+    /* We first have to assert that the client certificate provided can be traced back to the attached
+       CA chain in the EST configuration
+     */
+    const estConfig = await certificateTemplateService.getEstConfiguration({
+      isInternal: true,
+      certificateTemplateId
+    });
+
+    if (!estConfig.isEnabled) {
+      throw new BadRequestError({
+        message: "EST is disabled"
+      });
+    }
+
+    const leafCertificate = decodeURIComponent(sslClientCert).match(
```

Comment:
Cert-chain validation can be achieved with `@peculiar/x509` instead as in other comments in this review; using this module would reduce room for error with manual implementation.